### PR TITLE
Arcane Contributes and fixes some UI bits

### PR DESCRIPTION
**Pull request by ArcaneMusic**
My contributions:

Some alterations to the tgui elements of the Modsuit screen, makes having multiple UI-addative mods use a dropdown now, so you can show/hide info easier.

Improves UI layout of the radiation counter element.

Flashlight mod has it's own small, simple UI that lets you change the color of the flashlight as well as another button to toggle it.

Adds some returns to the construction steps switch of modsuits so they should construct properly.

Let me know what else needs tweaks.

### CODE REVIEW

**Comment by Fikou** [comment]
configurability shouldnt be done like this (sorry if i made it seem that way with how it currently does that in the modsuits ui_act)
it should be modular and dynamically generated in the tgui from add_configure_data() and received with configure_edit(key, value) which are already set up on the flashlight
the ui stuff on the flashlight should be removed and instead the dynamic generation of configuring menus should be added to the tgui

**Comment by Fikou** [comment]
this is nicer and thanks for the work, but the configging is still not "modular and dynamically generated in the tgui from add_configure_data() and received with configure_edit(key, value)"
please tell me if you will try to do that or if i should just merge this and do it myself

**Comment by Fikou** [comment]
this is nicer and thanks for the work, but the configging is still not "modular and dynamically generated in the tgui from add_configure_data() and received with configure_edit(key, value)"
let me know if you are interested in getting it to that state or if i should merge this in the current state and try to do it myself

File: code/modules/mod/mod_modules.dm [hunk 1]
```diff
@@ -721,14 +726,34 @@
...
+			while(!new_color)
+				new_color = input(user, "Choose a new color for [src]'s flashlight.", "Light Color",light_color) as color|null
+				if(!new_color)
+					return
+				if(color_hex2num(new_color) < 200) //Colors too dark are rejected
+					to_chat(user, "<span class='warning'>That color is too dark! Choose a lighter one.</span>")
+					new_color = null
+				light_color =  new_color
+				generate_worn_overlay()
+			return */
```
Comment (Fikou) [inline code/modules/mod/mod_modules.dm:756]: also whats with this comment

**Comment by ArcaneMusic** [comment]
Use what you can from this but again I think this was a more applicable way to doing this an what you're suggesting, it seems more like needless granularity

**Pull request closed without merging**